### PR TITLE
Add the Trino Snowflake connector story post.

### PR DESCRIPTION
**Pull request by bitsondatadev**
Adding the Snowlfake blog

### CODE REVIEW

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 1]
```diff
@@ -0,0 +1,294 @@
+---
+layout: post
+title:  "Integrating Trino and Snowflake: An open source success story"
+author: Brian Olsen
+excerpt_separator: <!--more-->
+canonical_url: https://bitsondata.dev/trino-snowflake-bloomberg-oss-win
```
Comment (bitsondatadev) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:6]: not published yet, make sure this link works before merging

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+upfront communication to validate ideas from newer contributors, and lack of
+communication by maintainers who see a large number of issues to address are
+two communication issues that stagnate a project. Maintainers are often doers
+that see more value in addressing quick-win work that flows from the
+well-established contributors of the project. Followthrough on either side can
+be difficult as newcomers don't want to be rude and maintainers accidentally
+forget or hope someone else will take the time to address the issues on that
+pull request. 
+
+![](https://i.snap.as/kbwKpWa7.webp)
```
Comment (bitsondatadev) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:88]: download image locally

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+support](https://github.com/trinodb/trino/pull/21365)(Apple) for the Snowflake
+integration. I love showcasing this trailblazing and yes, altruistic work Erik,
+Teng, Yuya, Martin, Manfred, and Piotr - and everyone who helped in the Trino
+community. A special thanks to the managers and leadership at Bloomberg and
+ForePaaS for their generous commitment of time and resources.
+
+As we celebrate this milestone, we're already looking forward to the next
+adventure. Here's to federating them all, together!
+
+================================================================================
```
Comment (bitsondatadev) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:289]:
```suggestion
```

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md (hunk 1 above)
Comment (yuuteng) [reply _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:6]: "This page is missing." During my test. Maybe because it's not published yet

**Comment by yuuteng** [comment]
Cool article. Good for me.

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+
+In the same way, demonstration of human cooperation is about more than just
+developing a connector; it's about the shared experiences, the friendships
+forged, and the skills honed in the pursuit of a common goal. The successful
+addition of the Trino Snowflake connector is a testament to the positive sum
+outcomes of open source collaboration. This journey has been about
+collaboration, learning, and growth that will benefit many.
+
+Another effect is that once the upfront hard work like this is done, it makes
+way for many valuable iterations like [adding Top-N
```
Comment (mosabua) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:276]:
```suggestion
many valuable iterations like [adding Top-N 
```

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+addition of the Trino Snowflake connector is a testament to the positive sum
+outcomes of open source collaboration. This journey has been about
+collaboration, learning, and growth that will benefit many.
+
+Another effect is that once the upfront hard work like this is done, it makes
+way for many valuable iterations like [adding Top-N 
+support](https://github.com/trinodb/trino/pull/21219)(Shoppee), [adding 
+Snowflake Iceberg REST catalog support](https://github.com/trinodb/trino/pull/21365)
+(Starburst), and [adding better type mapping 
+support](https://github.com/trinodb/trino/pull/21365)(Apple) for the Snowflake
```
Comment (mosabua) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:280]:
```suggestion
](https://github.com/trinodb/trino/pull/21365)(Apple) for the Snowflake
```

Comment (mosabua) [reply _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:280]: Need to make the preceding list into a coherent sentence

**Review by mosabua — CHANGES_REQUESTED** [review]

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+excerpt_separator: <!--more-->
+canonical_url: https://bitsondata.dev/trino-snowflake-bloomberg-oss-win
+---
+
+We’re seeing open source usher in a challenge to the economic model where the 
+success metric is increasing the commonwealth of economic capital. This 
+acceleration comes from playing positive-sum games with friends online and 
+avoiding limiting a community to a vision that only benefits a small number of 
+corporations or individuals. It’s hard to imagine how to embed such frameworks 
+within our current zero-sum winner-takes-all economic system. There’s certainly
```
Comment (colebow) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:14]: Brian Olsen takes down capitalism, one open source blog post at a time (I do like that you're not mincing words here)

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+We’re seeing open source usher in a challenge to the economic model where the 
+success metric is increasing the commonwealth of economic capital. This 
+acceleration comes from playing positive-sum games with friends online and 
+avoiding limiting a community to a vision that only benefits a small number of 
+corporations or individuals. It’s hard to imagine how to embed such frameworks 
+within our current zero-sum winner-takes-all economic system. There’s certainly 
+no shortage of heated debates around how to construct a harmonious relationship 
+between the open source community and companies participating in them. Something
+we don’t talk about enough are the positive examples of when a coordinated 
+effort in open source sticks the landing and so many benefit from it.
```
Comment (colebow) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:18]:
```suggestion
effort in open source sticks the landing, and so many benefit from it.
```

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+This post highlights the extraordinary contributions of 
+[Erik Anderson](https://www.linkedin.com/in/erikanderson/), 
+[Teng Yu](https://www.linkedin.com/in/tyu-fr/), 
+[Yuya Ebihara](https://www.linkedin.com/in/ebyhr/), and the broader 
+[Trino community](https://github.com/trinodb/trino) to finally contribute the 
+long-coveted 
+[Trino Snowflake Connector](https://trino.io/docs/current/connector/snowflake.html).
+
+It is both a success story and some wisdom for corporations using and providing
+services around open source should be involved in open source . This provides a
```
Comment (colebow) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:29]:
```suggestion
services around open source should be involved in open source. This provides a
```

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+blue print for those who want to contribute to open source the types of
```
Comment (colebow) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:30]:
```suggestion
blueprint for those who want to contribute to open source the types of
```

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+devex), which aims to improve both the user and contributor experiences in the
+developer community by
+[empower decision making by doing](https://en.wikipedia.org/wiki/Experiential_learning),
+[removing inefficiencies](https://trino.io/blog/2023/01/09/cleaning-up-the-trino-backlog),
+and as we'll cover here, exposing untapped opportunities. 
+
+Much like any open source project, maintainers on the Trino project struggle at
+communicating our lack of proper resources to build and test new features built
+on various proprietary software. For those less familiar, Trino is a federated
+query engine with
```
Comment (colebow) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:52]: I think we can probably leave out the brief explainer for what Trino is for the version of the blog going on the Trino website?

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 11]
```diff
@@ -0,0 +1,294 @@
...
+[Phillipe Gagnon](https://www.linkedin.com/in/pfgagnon), a similar pattern
+emerged [with the second pull request](https://github.com/trinodb/trino/pull/10387)
+where the development velocity started strong and after some months stagnated.
+
+### Cognitive surplus and communication deficit
+
+One of the most unfortunate repeated issues I see in the open source projects
+I've contributed to and helped maintain is that various well-known larger
+objectives known among the core group often move so fast, while individual
+updates that don't fit in a larger project narrative has higher likelihood to
```
Comment (colebow) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:71]:
```suggestion
updates that don't fit in a larger project narrative have a higher likelihood to
```

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 12]
```diff
@@ -0,0 +1,294 @@
...
+families, and self care, that they dedicate most of their remaining energy to
+ensuring they write quality code and tests to the best of their ability. Lack of
+upfront communication to validate ideas from newer contributors, and lack of
+communication by maintainers who see a large number of issues to address are
+two communication issues that stagnate a project. Maintainers are often doers
+that see more value in addressing quick-win work that flows from the
+well-established contributors of the project. Followthrough on either side can
+be difficult as newcomers don't want to be rude and maintainers accidentally
+forget or hope someone else will take the time to address the issues on that
+pull request.
```
Comment (colebow) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:86]: I think it's also worth noting that there generally isn't a great process in place to handle these on-off contributions from individuals? Like, even for aggressively open core companies, engineers' jobs so rarely involve a requirement/incentive/reward to facilitate and help out with non-roadmap open source PRs.

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 13]
```diff
@@ -0,0 +1,294 @@
...
+benefit from your small and significant change floods your mind and you feel
+like you’ve improved humanity just that one little bit more. 
+
+Maintainers are in a constant state of pulling triage on all the surplus of
+innovation being thrown at them and simultaneously trying to look for more help
+reviewing and being the expert at some areas of the code. As you can imagine,
+good communication can be hard to come by as many newcomers are strangers and
+concerned they are wasting precious time by asking too many questions rather
+than just showing a proof of concept. This backfires for multiple reasons that
+go outside of the scope of this post.
```
Comment (colebow) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:103]: I think you could briefly touch on them if you wanted, this feels a little abrupt...

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 14]
```diff
@@ -0,0 +1,294 @@
...
+[countless number of updates](https://github.com/trinodb/trino/pull/17909#issuecomment-1841809727)
+you have to make to your changes to accommodate the ever changing project]
+shifting beneath your feet as you simultaneously try to make progress on
+[suggestions from those reviewing your code](https://github.com/trinodb/trino/pull/17909#pullrequestreview-1793724311).
+
+Many critics of open source like to point this out as a drawback of this model,
+when in fact, this same problem exists in closed-source systems, but closed
+source projects can generally delay difficult decisions to make fast upfront
+progress to meet certain deadlines. This may be seen as an advantage at first,
+but as many developers can attest, this simply leads to technical debt and
```
Comment (colebow) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:184]: 👍 love this

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 15]
```diff
@@ -0,0 +1,294 @@
...
+Despite how many problems Trino already solves for them, Bloomberg ideally still
+wants to maintain as few things as possible to relieve their engineer’s time.
+Luckily, open source projects are generally more than happy to accept features
+that the community all benefit from. This doesn’t mean we shouldn’t appreciate
+when companies contribute. This generosity and forward-thinking approach enabled
+Erik and Teng to combine their battle-tested connectors, crafting a high value
+creation for the community.
+
+Yet another issue that has come up much more recently is the
+[XZ exploit](https://www.darkreading.com/application-security/xz-utils-scare-exposes-hard-truths-in-software-security)
```
Comment (colebow) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:229]: I'm not sure the tangent of security exploits really belongs in this blog post - or at least it isn't an issue that the Trino community would be super concerned with, I think.

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 16]
```diff
@@ -0,0 +1,294 @@
...
+## Esprit de Corps
+
+The marines use the mantra, “Esprit de Corps,” latin for “spirit of the people”,
+which I mistakenly took the “Corps” part for the Marine Corps rather than the
+more general meaning of a body or group of people. In fact, it expresses [the 
+common spirit existing in the members of a group and inspiring enthusiasm,
+devotion, and strong regard for the honor of the 
+group.](https://www.merriam-webster.com/dictionary/esprit%20de%20corps) Any time
+I see this type of shared and selfless cooperation in open source, I’m reminded
+of the bond, friendships, and care of me and my fellow marines in times of war.
```
Comment (colebow) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:262]:
```suggestion
of the bond, friendships, and care of me and my fellow marines.
```
I think removing this makes the sentence more powerful

**Review by colebow — COMMENTED** [review]
Great post. I have some comments, but feel free to disregard any of them, I think it's more important that it's clearly in your voice than polished to a T

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 17]
```diff
@@ -0,0 +1,294 @@
...
+
+This post highlights the extraordinary contributions of 
+[Erik Anderson](https://www.linkedin.com/in/erikanderson/), 
+[Teng Yu](https://www.linkedin.com/in/tyu-fr/), 
+[Yuya Ebihara](https://www.linkedin.com/in/ebyhr/), and the broader 
+[Trino community](https://github.com/trinodb/trino) to finally contribute the 
+long-coveted 
+[Trino Snowflake Connector](https://trino.io/docs/current/connector/snowflake.html).
+
+It is both a success story and some wisdom for corporations using and providing
```
[the remark this answers is not in the document]
Comment (bitsondatadev) [reply _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:28]: Despite the importance of delivering marketing and education in a community (aka
[edutainment](https://en.wikipedia.org/wiki/Educational_entertainment)), it’s
only the first part of the equation of what makes open source projects successful.

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 18]
```diff
@@ -0,0 +1,294 @@
...
+[edutainment](https://en.wikipedia.org/wiki/Educational_entertainment)), it’s
+only the first part of the equation of what I aim to do as a developer advocate.
+Once developers  see some exciting video or tutorial, they ultimately land on
+the docs site, GitHub, StackOverflow, or some communication platform in the
+community. It's at this point that developers can easily lose the motivation if
+the the community and docs lacks proper getting started materials or worse, a
+silent community. This is how I categorize the developer experience (aka
+devex), which aims to improve both the user and contributor experiences in the
+developer community by
+[empower decision making by doing](https://en.wikipedia.org/wiki/Experiential_learning),
```
[the remark this answers is not in the document]
Comment (bitsondatadev) [reply _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:45]: empowering decisions through hands-on learning

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 19]
```diff
@@ -0,0 +1,294 @@
...
+
+It became apparent that each time there was 
+[a discussion](https://github.com/trinodb/trino/pull/2551#issuecomment-709220790)
+for how to do
+[integration testing](https://github.com/trinodb/trino/pull/10387#issuecomment-1008430060)
+there was no good way to test a Snowflake instance with the lack of funding for
+the project. Trino has a high bar for quality and none of the maintainers felt
+it was a risk worth taking due to the likely popularity of the integration and
+likelihood of future maintenance issues. Once each pull request hit this same
+fate, it stalled with nobody really knowing how to resolve the real issue of
```
[the remark this answers is not in the document]
Comment (bitsondatadev) [reply _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:115]: I'll make this a weaker statement, but this is my account of why these kept stalling.

> Once each pull request hit this same
fate, it stalled with no clear path to resolve the real issue of funding the
Snowflake infrastructure needed by the
[Trino Software Foundation (TSF)](https://trino.io/foundation.html). It’s never
fun to mention that you can’t move forward on work with constraints like these,
and without a monetary solution, silence is what is experienced on the side of
the contributor

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 20]
```diff
@@ -0,0 +1,294 @@
...
+limited financial resources of the
+[Trino Software Foundation (TSF)](https://trino.io/foundation.html). It’s never
+fun to mention that you can’t move forward on work with constraints like these
+and yet nobody has the time to look into a monetary solution.
+
+Noticing that Teng had already done a significant amount of work to contribute
```
[the remark this answers is not in the document]
Comment (bitsondatadev) [reply _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:121]: I link to all their LinkedIn profiles above.

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 21]
```diff
@@ -0,0 +1,294 @@
...
+"Oh I wish I would have known that's the problem, the solution is simple,
+Bloomberg will provide the TSF a Snowflake account."
+
+Done!
+
+Just as in business, **you can never underestimate the power of communication in
+an open source project** as well. Shortly after Erik, Teng, and I discussed the
+best ways to merge their work, set up the Snowflake accounts for Trino
+maintainers, and start the arduous process of building a thorough test suite
+with the help of Yuya, Piotr, Manfred, and Martin.
```
[the remark this answers is not in the document]
Comment (bitsondatadev) [reply _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:142]: Already did Yuya above, but I can do full names for everyone else w/ LinkedIn links

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md (hunk 19 above)
[the remark this answers is not in the document]
Comment (mosabua) [reply _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:115]: fair enough

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md (hunk 6 above)
Comment (mosabua) [reply _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:14]: haha .. currently reading https://en.wikipedia.org/wiki/Chokepoint_Capitalism myself ..

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md (hunk 10 above)
Comment (bitsondatadev) [reply _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:52]: It's going to be posted to my site and I want it to serve as a reference point for anyone I discuss with.

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md (hunk 12 above)
Comment (bitsondatadev) [reply _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:86]: That feels like another blog post for me...I have a lot to say there haha

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md (hunk 13 above)
Comment (bitsondatadev) [reply _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:103]:
```
This backfires when developers will spend
a large portion of their time developing a solution that is not compatible with the
project, and maintainers will lose the opportunity to quickly spin up on the value
of the new feature. This is why regular
[contributor meetings](https://github.com/trinodb/trino/wiki/Contributor-meetings)
help solve both of these issues synchronously to cut out the delayed feedback loops.
```

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 22]
```diff
@@ -0,0 +1,307 @@
...
+success metric is increasing the commonwealth of economic capital. This 
+acceleration comes from playing positive-sum games with friends online and 
+avoiding limiting a community to a vision that only benefits a small number of 
+corporations or individuals. It’s hard to imagine how to embed such frameworks 
+within our current zero-sum winner-takes-all economic system. There’s certainly 
+no shortage of heated debates around how to construct a harmonious relationship 
+between the open source community and companies participating in them. Something
+we don’t talk about enough are the positive examples of when a coordinated 
+effort in open source sticks the landing, and so many benefit from it.
+
```
Comment (mosabua) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:21]: maybe here

**Review by colebow — APPROVED** [review]
Looks ready - just needs a date update. We've also got a Trino Fest blog going up soon, maybe we should discuss on Slack to figure out the perfect timing

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 23]
```diff
@@ -0,0 +1,307 @@
...
+fate, it stalled with no clear path to resolve the real issue of funding the
+Snowflake infrastructure needed by the
+[Trino Software Foundation (TSF)](https://trino.io/foundation.html). It’s never
+fun to mention that you can’t move forward on work with constraints like these,
+and without a monetary solution, silence is what is experienced on the side of
+the contributor.
+
+Noticing that Teng had already done a significant amount of work to contribute
+his Snowflake connector, I reached out to him to see if we could brainstorm a
+solution. Not long after, Erik Anderson also reached out to get my thoughts on
```
Comment (mosabua) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:132]:
```suggestion
solution. Not long after, Erik also reached out to get my thoughts on
```

File: _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md [hunk 24]
```diff
@@ -0,0 +1,307 @@
...
+[Trino version 440](https://trino.io/docs/current/release/release-440.html#general), 
+and made connection to the most widely adopted cloud warehouse possible.
+
+Once the hard work was done, many valuable iterations like 
+[adding Top-N support](https://github.com/trinodb/trino/pull/21219)(Shoppee),
+[adding Snowflake Iceberg REST catalog support](https://github.com/trinodb/trino/pull/21365)
+(Starburst), and [adding better type 
+mapping](https://github.com/trinodb/trino/pull/21365)(Apple) were added to the
+Snowflake integration. I love showcasing this trailblazing and yes, altruistic
+work Erik, Teng, Yuya, Martin, Manfred, and Piotr - and everyone who helped in
```
Comment (mosabua) [inline _posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md:296]:
```suggestion
work from Erik, Teng, Yuya, Martin, Manfred, and Piotr - and everyone who helped in
```

**Review by mosabua — APPROVED** [review]
Looks good to go now. Some minor nits to fix. Before we merge we just gotta change to the appropriate date.

**Comment by bitsondatadev** [comment]
Done!

**Review by mosabua — CHANGES_REQUESTED** [review]
Just looking at this again to see where things are at. I think we need to avoid the off-brand usage of a bunny to symbolize Commander Bun Bun unless we are happy with anybody doing that out there. Personally I think we should make sure CBB stays recognizable.

In terms of the content itself, I am still not sure how much on brand this should be vs how much it can be a personal view and perspective. In the later case it probably has to be made more explicit as a guest post or something like that. 

All that in mind I am not sure how to best proceed. Not merge but advertise your original post on Trino channels? Modify and simplify and include explicit wording and link about the "full story from Brians perspective" and then advertise it? 

I think @martint @dain @electrum and also @bitsondatadev should chime in on what is their suggested next step and preferences are here so we can proceed.

**Comment by mosabua** [comment]
Closing since this lost momentum and is not relevant as a current news update anymore.

**Pull request closed without merging**